lib/analyzer/types.ts

```diff
@@ -77,6 +77,7 @@ export interface StaticAnalysis {
   autoDetectedUserInstructions?: DockerFileAnalysis;
   applicationDependenciesScanResults: AppDepsScanResultWithoutTarget[];
   manifestFiles: ManifestFile[];
+  imageLabels?: { [key: string]: string };
```

Comment:
why is this optional?

Reply:
Hi Amir 👋 
It's because we only extract labels from images based on docker-archive, and not also oci-archive based images which are only supported in cli. (We have just decided that we are taking this trade-off, you can see conversation [here](https://snyk.slack.com/archives/C01EPSMGZML/p1612885007094500?thread_ts=1611595377.023500&cid=C01EPSMGZML))

Reply:
that's okay, and thanks for the context 🙏
but why have the property optional, and not just leave it empty when it's "off"?

Reply:
Sure, that would also work.
If you prefer it I can change now, or we change it later when we add oci-archive support.

Reply:
now is preferable please 🙏

Reply:
I'm missing something in here probably, but why is empty object better than `undefined`? Isn't `undefined` more explicit and easier to check later when we're creating the fact, so if we change to empty object we have to modify this: 
```
 if (depsAnalysis.imageLabels)
 ```
 To be more complicated checking if the object is empty?

Reply:
it's probably more of a style choice rather than better/worse.

in my opinion, being very explicit and deterministic about what our types look like is preferable.
```undefined``` is the opposite of explicit because if it's ```undefined``` then it would just be missing from the object entirely instead of being present with an empty value.

I prefer "actively" knowing than no image labels were associated with a payload over having it implicitly hidden.
that's just my style though!

Reply:
In that specific case, I think it's actually better, as we don't really need that conditional.. and can just iterate over labels. Hence simpler and cleaner code.

It's a small change with different trade-offs... imho you can argue on both sides. :)

Reply:
I see I see! Thanks Amir! You're right, I like the explanation! I guess coming from the c# world I'm very used to/like nullable props and we do use that approach more in the plugin (I would have to confirm that information hehe) but it's great to consider different approaches :-)